projects/plugins/jetpack/class.jetpack.php

```diff
@@ -1737,14 +1737,22 @@ public static function is_active() {
 	 * users and this option will be available by default for everybody.
 	 *
 	 * @since 9.6.0
+	 * @since 9.7.0 returns is_connected in all cases and adds filter to the returned value
 	 *
 	 * @return bool is the site connection ready to be used?
 	 */
 	public static function is_connection_ready() {
-		if ( ( new Status() )->is_no_user_testing_mode() ) {
-			return self::connection()->is_connected();
-		}
-		return (bool) self::connection()->has_connected_owner();
+		/**
+		 * Allows filtering whether the connection is ready to be used. If true, this will enable the Jetpack UI and modules
+		 *
+		 * Modules will be enabled depending on the connection status and if the module requires a connection or user connection.
+		 *
+		 * @since 9.7.0
+		 *
+		 * @param bool                                  $is_connection_ready Is the connection ready?
+		 * @param Automattic\Jetpack\Connection\Manager $connection_manager Instance of the Manager class, can be used to check the connection status.
+		 */
+		return apply_filters( 'jetpack_is_connection_ready', self::connection()->is_connected(), self::connection() );
```

Comment:
Nice, thanks for adding this filter here!

Do you think we should handle Atomic here explicitly or depend on the filter? I'm leaning towards the second.

Reply:
I agree we should let them filter this if needed via wpcomsh